src/components/Modal/Modal.tsx

```diff
@@ -149,19 +258,42 @@ export function useModal(): any {
       bodyContent,
       closeButtonLabel,
       headingText,
+      confirmButtonLabel,
+      onCancel,
+      onConfirm,
+      onClose,
+      type,
       id,
       ...rest
     }: React.PropsWithChildren<BaseModalProps>) => {
       return (
-        <BaseModal
-          bodyContent={bodyContent}
-          closeButtonLabel={closeButtonLabel}
-          headingText={headingText}
-          id={id}
-          isOpen={isOpen}
-          onClose={onClose}
-          {...rest}
-        />
+        <>
+          {isDefaultType(type) ? (
+            <BaseModal
+              bodyContent={bodyContent}
+              headingText={headingText}
+              id={id}
+              type={type}
+              isOpen={isOpen}
+              onClose={onClose}
```

Comment:
The more I look at this, the more it's confusing and not liking some of the decisions I made before. This is using the local `onClose` from the component, but there is also an `onClose` in the outer scope and this can be confusing.

In the storybook documentation, there's a `useModal` story that is using the "default" type. Can you add another storybook story example for `useModal` that uses the "confirmation" type?

Reply:
I would dearly love to change the `onClose` in the outer scope– if we're already full steam ahead w `type` do you think i could? The only reason i didn't was that i didn't want to create breaking changes. 
useModal example added but let me know if that's where it should be placed in the doc.

Reply:
As we talked in our chat with Marty, do not rename the `onClose` function that consuming apps use. But if you want, you can update the internal `onClose` on line 250 as `onClose: [someOtherName]`. You'd just need to also export it on line 296 as `onClose: [someOtherName]`.

Reply:
For that case, the dev using the DS won't see the renamed function since it's internal. So it could be updated but won't be any less confusing.